src/app/frontend/replicasetlist/replicasetlist_state.js

```diff
@@ -32,15 +33,32 @@ export default function stateConfig($stateProvider) {
       replicaSets: resolveReplicaSets,
     },
     templateUrl: 'replicasetlist/replicasetlist.html',
+    onEnter: initState,
   });
 }
 
 /**
+ * Avoids entering replica set list page when there are no replica sets.
+ * Used f.e. when last replica set gets deleted.
+ * Transition to: zerostate
+ * @param {!ui.router.$state} $state
+ * @param {!angular.$timeout} $timeout
+ * @param {!backendApi.ReplicaSetList} replicaSets
+ * @ngInject
+ */
+function initState($state, $timeout, replicaSets) {
```

Comment:
`initState`? This function does not do any initialization :) It redirects to zero state when there are no replica sets. Apply name accordingly